Ports the Plasteel Arming Sword from Frontier Station

AeraAuling opened the pull request:
<!-- If you are new to the Delta-V repository, please read the [Contributing Guidelines](https://github.com/DeltaV-Station/Delta-v/blob/master/CONTRIBUTING.md) -->

## About the PR
ports the Plasteel Arming Sword from Frontier Station to Delta-V, and makes it available in the Salvage Points Shop for 500 points, or in cargo for 2500 for 2, in a secure weapons crate for security

(original PR by MagnusCrowe at https://github.com/new-frontiers-14/frontier-station-14/pull/887)

## Why / Balance
Simply adds a new sword. It's relatively fast, only does 15 damage, but it's a melee weapon that isn't either a Big Hammer or some ramshackle thing you made out of cables and sticks. Also, it looks pretty neat to hold. Admittedly, I think it just uses edited claymore sprites though so I will probably resprite it at some point.

## Technical details
Entirely YAML changes

## Media
<!-- Attach media if the PR makes ingame changes (clothing, items, features, etc).
Small fixes/refactors are exempt. -->

## Requirements
<!-- Confirm the following by placing an X in the brackets [X]: -->
- [X] I have tested all added content and changes.
- [X] I have added media to this PR or it does not require an ingame showcase.
<!-- You should understand that not following the above may get your PR closed at maintainer’s discretion -->

## Breaking changes
<!-- List any breaking changes, including namespaces, public class/method/field changes, prototype renames; and provide instructions for fixing them. -->

**Changelog**
<!-- Add a Changelog entry to make players aware of new features or changes that could affect gameplay.
Make sure to read the guidelines and take this Changelog template out of the comment block in order for it to show up.
Changelogs must have a :cl: symbol, so the bot recognizes the changes and adds them to the game's changelog. -->
:cl:
- add: NT has started to supply stations with high-quality arming swords. Security and Salvage can pick one up from cargo for their work. (credit to MagnusCrowe)

AeraAuling committed:
adds the plasteel arming sword from frontier

AeraAuling committed:
Merge branch 'DeltaV-Station:master' into swords

AeraAuling committed:
forgot to change the price back

AeraAuling committed:
Merge branch 'swords' of https://github.com/AeraAuling/Delta-v-stuff into swords

deltanedas commented on Resources/Locale/en-US/_DV/prototypes/catalog/cargo/cargo-armory.ftl:
```diff
@@ -12,3 +12,6 @@ ent-ArmoryEnergyGun = { ent-CrateArmoryEnergyGun }
     .desc = { ent-CrateArmoryEnergyGunMini.desc }
+
+ent-ArmoryArmingSword = { ent-CrateArmoryArmingSword }
+    .desc = { ent-CrateArmoryArmingSword.desc}
```
remove

deltanedas commented on Resources/Locale/en-US/_DV/prototypes/catalog/fills/crates/armory-crates.ftl:
```diff
@@ -12,3 +12,6 @@ ent-CrateArmoryEnergyGun = Energy Gun Crate
     .desc = Contains three Miniature Energy Guns. Requires Armory access to open.
+
+ent-CrateArmoryArmingSword = Arming Sword Crate
+    .desc = Contains two Plasteel Arming Swords. Requires Armory access to open.
```
use `name: ` and `description: ` on the entity prototype

deltanedas commented on Resources/Prototypes/_DV/Catalog/Fills/Crates/armory.yml:
```diff
@@ -30,21 +30,30 @@
+  components:
+  - type: StorageFill
+    contents:
+      - id: PlasteelArmingSword
+        amount: 2
```
unindent by 2 spaces

deltanedas commented on Resources/Prototypes/_NF/Entities/Objects/Weapons/Melee/sword.yml:
```diff
@@ -0,0 +1,32 @@
+    slots:
+    - back
+    - suitStorage
+    - belt
```
almost all of these fields are inherited from captains sabre, remove duplicate components and fields with the same values

AeraAuling committed:
hope I did this right

AeraAuling commented:
I _think_ I did it all right

deltanedas commented on Resources/Prototypes/_NF/Entities/Objects/Weapons/Melee/sword.yml:
```diff
@@ -0,0 +1,26 @@
+  id: PlasteelArmingSword
+  description: An ancient design manufactured with modern materials and machines for a very specific target demographic.
+  components:
+  - type: Sharp
```
still duplicated

deltanedas commented on Resources/Prototypes/_NF/Entities/Objects/Weapons/Melee/sword.yml:
```diff
@@ -0,0 +1,26 @@
+# original PR by MagnusCrowe at https://github.com/new-frontiers-14/frontier-station-14/pull/887
+
+- type: entity
+  name: plasteel arming sword
```
move this below the `id:` line

deltanedas commented on Resources/Prototypes/_NF/Entities/Objects/Weapons/Melee/sword.yml:
```diff
@@ -0,0 +1,26 @@
+  - type: Sharp
+  - type: Sprite
+    sprite: _NF/Objects/Weapons/Melee/armingsword.rsi
+    state: icon
```
remove the state line its inherited

deltanedas commented on Resources/Prototypes/_NF/Entities/Objects/Weapons/Melee/sword.yml:
```diff
@@ -0,0 +1,26 @@
+    state: icon
+  - type: MeleeWeapon
+    wideAnimationRotation: -135
+    attackRate: 1.5
```
these attack rate is inherited

AeraAuling committed:
Merge branch 'master' into swords

AeraAuling committed:
this time maybe

AeraAuling committed:
Merge branch 'swords' of https://github.com/AeraAuling/Delta-v-stuff into swords

AeraAuling committed:
Merge branch 'master' into swords

ThataKat commented:
Direction is reviewing, get back to you in a day or so

deltanedas committed:
:trollface:

Signed-off-by: deltanedas <39013340+deltanedas@users.noreply.github.com>

deltanedas commented on Resources/Textures/_NF/Objects/Weapons/Melee/armingsword.rsi/meta.json:
```diff
@@ -0,0 +1,30 @@
+        "name": "equipped-BELT",
+        "directions": 4
+      }
+    ]
```
fix the indent, 4 spaces. dont mix and match

AeraAuling committed:
and that too maybe

AeraAuling committed:
Merge branch 'swords' of https://github.com/AeraAuling/Delta-v-stuff into swords

AeraAuling committed:
Merge branch 'master' into swords

AeraAuling committed:
Merge branch 'DeltaV-Station:master' into swords

deltanedas reviewed (changes requested):
4 spaces not 2...

dvir001 commented:
Please consider mentioning the original PR, and possibly adding the original maker to changelog.

ThataKat commented:
Direction here,

We'd prefer it not be in the salvage points shop to prevent people buying them en masse. If you have another suggestion for where to put them so salvage can obtain them we'd love to hear it (I suppose they could settle for ordering them and asking sec to open it). Besides that, core concept is approved.

fow111 commented:
> Direction here,
> 
> We'd prefer it not be in the salvage points shop to prevent people buying them en masse. If you have another suggestion for where to put them so salvage can obtain them we'd love to hear it (I suppose they could settle for ordering them and asking sec to open it). Besides that, core concept is approved.

Can't salvage buy crushers enmasse too? What's the difference between a ton or crushers and a ton of swords?

AeraAuling committed:
remove reflect chance I think

Signed-off-by: AeraAuling <133451603+AeraAuling@users.noreply.github.com>

AeraAuling commented:
> Please consider mentioning the original PR, and possibly adding the original maker to changelog.

I had put the original PR into the sword.yml file, but I've added it to the post as well and added their name to the changelog. Also, reflect chance removed by request

AeraAuling committed:
Merge branch 'DeltaV-Station:master' into swords

Avalon-Proto commented:
I do agree with Aera, why not let salvage have the swords. When they can already get crushers that are a million times better. The swords are just fun flavor :blunt:

fow111 commented:
If salvage buying plasteel swords en masse is an issue then upoing the price to something like 5000 points could also work, which is definitely not worth the price but it would be more of a prestige thing :P

AeraAuling committed:
Merge branch 'master' into swords

AeraAuling commented:
quick bump for direction, reflect chance was removed as discussed in discord

AeraAuling committed:
Merge branch 'master' into swords

AeraAuling committed:
fix indents maybe, I dunno why this part in particular is so difficult for me I just can't SEE right

Signed-off-by: AeraAulin <133451603+AeraAuling@users.noreply.github.com>

AeraAuling committed:
Merge branch 'master' into swords

Lyndomen committed:
Merge branch 'master' into swords

AeraAuling commented:
test fail seems to be uh... Trystopnukeops fromconstantlyfailing. never touched those files so unsure how real that fail is

AeraAuling committed:
Merge branch 'master' into swords

AeraAuling committed:
Merge branch 'master' into swords

deltanedas committed:
reformat meta.json

Signed-off-by: deltanedas <39013340+deltanedas@users.noreply.github.com>
